gordo_components/server/server.py

```diff
@@ -123,10 +133,77 @@ def base_healthcheck():
     return app
 
 
-def run_server(host: str = "0.0.0.0", port: int = 5555, debug: bool = False):
-    app = build_app()
-    app.run(host, port, debug=debug, threaded=False)
+def run_server(
+    host: str, port: int, workers: int, worker_connections: int, log_level: str
+):
+    """
+    Run application with Gunicorn server using Gevent Async workers
+
+    Parameters
+    ----------
+    host: str
+        The host to run the server on.
+    port: int
+        The port to run the server on.
+    workers: int
+        The number of worker processes for handling requests.
+    worker_connections: int
+        The maximum number of simultaneous clients per worker process.
+    log_level: str
+        The log level for the server.
+    """
 
+    try:
+        ipaddress.ip_address(host)
+    except Exception:
+        raise
+
+    if not (1 <= port <= 65535):
+        raise ValueError(f"{port} is not a valid port number")
+
+    gunicorn_log_levels = Logger.LOG_LEVELS.keys()
+    if log_level not in gunicorn_log_levels:
+        raise ValueError(
+            f"'{log_level}' log level is not support by Gunicorn. Must be one of {', '.join(gunicorn_log_levels)}."
+        )
+
+    if workers <= 0:
+        raise ValueError("Workers must be greater than 0, not {workers}.")
+
+    if worker_connections <= 0:
+        raise ValueError("Workers must be greater than 0, not {workers}.")
+
+    cmd = [
+        "gunicorn",
+        "--bind",
+        f"{host}:{port}",
+        "--worker-class",
+        "gevent",
+        "--access-logfile",
+        "-",
+        "--error-logfile",
+        "-",
```

Comment:
Not worth changing, but I suppose `-` for the log files indicates outputting to stdout? Is this not the default behavior?

Reply:
Removing these here unnecessary args.